### PR TITLE
Add initial GStreamer integration and streamline the UI

### PR DESCRIPTION
**Pull request by sdroege**
@GuillaumeGomez 

I would create a tag for the "Stream-line UI" commit and make this the very first step of the workshop. And then the GStreamer integration commit the second step.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -43,109 +58,193 @@ impl AppWeak {
...
         p.set_modal(true);
-        p.set_program_name("RustFest GTK+GStreamer");
+        p.set_program_name("RustFest 2018 GTK+ & GStreamer WebCam Viewer");
+
+        // When any response on the dialog happens, we simply destroy it.
+        //
+        // We don't have any custom buttons added so this will only ever
+        // handle the close button, otherwise we could distinguish the
+        // buttons by the response
+        p.connect_response(|dialog, _response| {
```
Comment (GuillaumeGomez) [inline src/main.rs:97]: You sure about this? What happens when you click on other buttons than "close"?

Comment (sdroege) [reply src/main.rs:97]: Which other buttons?

Comment (GuillaumeGomez) [reply src/main.rs:97]: There are three buttons on an about dialog, no?

Comment (sdroege) [reply src/main.rs:97]: The close button, and the credits/about "tab" buttons. Only the first causes a response, and any other buttons one would add

Comment (GuillaumeGomez) [reply src/main.rs:97]: Good to know. :)

**Comment by GuillaumeGomez** [comment]
When trying to run it on osx, I get:

```
cargo run --release
    Finished release [optimized] target(s) in 0.09s
     Running `target/release/workshop`

(<unknown>:16856): Gtk-WARNING **: 16:36:06.063: Locale not supported by C library.
	Using the fallback 'C' locale.

(<unknown>:16856): GStreamer-CRITICAL **: 16:36:06.161: gst_bin_add: assertion 'GST_IS_ELEMENT (element)' failed

(<unknown>:16856): GStreamer-CRITICAL **: 16:36:06.161: gst_element_get_static_pad: assertion 'GST_IS_ELEMENT (element)' failed

(<unknown>:16856): GStreamer-CRITICAL **: 16:36:06.161: gst_object_unref: assertion 'object != NULL' failed
Error: Error { domain: gst_parse_error, code: 1, message: "pas d’élément «\u{a0}queue\u{a0}»" }
```

I installed the last `gstreamer` version just before and followed OSX instructions (even the export one) but I assume I still forgot something... Is there anything else needed to be installed in order to run it?

**Comment by sdroege** [comment]
Run G_DEBUG=fatal_warnings in gdb and get a backtrace of the first error. You're maybe missing the gtksink from gst-plugins-good

**Comment by sdroege** [comment]
I'll add better error reporting later

**Comment by sdroege** [comment]
For homebrew you need to pass --enable-gtk3 to the gst-plugins-good build.

If you use the GStreamer binaries from gstreamer.freedesktop.org, they don't contain any GTK support right now.

**Comment by sdroege** [comment]
Ah the french there means "no such element 'queue'". Your GStreamer setup is completely broken then :) How did you install things?

**Comment by GuillaumeGomez** [comment]
I followed gstreamer-rs README. :'(

**Comment by sdroege** [comment]
There are two options for macos. Which one did you take?

Best would be for now to use homebrew, and make sure to enable gtk support. Once you have that and if it still does not work, run `gst-inspect-1.0 queue` and `gst-inspect-1.0 gtksink`, `which gst-inspect-1.0`, `GST_DEBUG=6 gst-inspect-1.0 &> log` and paste the output and the log file. We take it from there

**Pull request merged**